week3/index.js

```diff
@@ -1,15 +1,77 @@
-const Express = require('express')
-const bodyParser = require('body-parser')
+"use strict"
+const express = require("express");
+const app = express();
+const bodyParser = require("body-parser");
+const fs = require("fs");
+const helperFunctions = require("./helperFunctions.js")
 
-const app = Express()
 
-app.use(bodyParser.json())
+app.use(bodyParser.urlencoded({ extended: false }));
 
-const {list, create, update, remove} = require('./actions')
 
-app.get('/todos', list)
-app.post('/todos', create)
-app.put('/todos/:id', update)
-app.delete('/todos/:id', remove)
+app.get("/", (req, res) => {
+    helperFunctions.showHelpMenu()
+        .then(data => {
+            req.body = data
+            res.send(req.body + "\n")
+        })
+        .catch(err => res.send(err.message))
+})
 
-app.listen(3000)
\ No newline at end of file
+app.get("/help", (req, res) => {
+    helperFunctions.showHelpMenu()
+        .then(data => {
+            req.body = data
+            res.send(req.body + "\n")
+        })
+        .catch(err => res.send(err.message))
+
+})
+
+app.get("/list", (req, res) => {
+    helperFunctions.readFileFunction("to-do-list.json")
+        .then((data) => {
+            data = data || "[]";
+            helperFunctions.myTodoList = JSON.parse(data);
+            if (helperFunctions.myTodoList.length === 0) {
+                res.send("You have no tasks to do, you are free\n");
+            } else {
+                res.send("Tasks to do: " + JSON.stringify(helperFunctions.myTodoList) + "\n");
+            }
+        })
+        .catch(err => res.send(`an error happened while reading the file ${err} \n`));
+})
+
+app.post("/add/:task", (req, res) => {
+    helperFunctions.addingTask(req.params.task, "to-do-list.json")
+        .then(data => {
+            res.send("A task has been added \n")
+        })
+        .catch(err => res.send("failed to add the task " + err.message + "\n"))
+})
+
+app.put("/update/:id/:updatedTask", (req, res) => {
+    let taskId = req.params.id;
+    let newTask = req.params.updatedTask;
+    helperFunctions.updateFunction(taskId, newTask)
+        .then(() => res.send("Task has been updated\n"))
```

Comment:
Nice job with these functions!

A common pattern you'll see for APIs that deal mostly with JSON is to send back a short JSON response, rather than a string like you're doing here. This can be simple, like:
```
{ "result": "success" }
```
And then if there's an error, you can send back:
```
{ "result": "error", "errorMessage": err.message }
```